Updated links/buttons based on help page, helpful error message

emhastings opened the pull request:
The Blocky documentation is done, and this commit updates the helpURLs for the blocks reflecting this.  It also changes the appearance of the submit button.  It still does not actually submit
anything-- Jaime, I think you said last time we met that you were going to work on this?

When we're done with that, we also have to remember to put a download link for the Blockly stuff on the Google site with the other downloads, and probably put our local version up on cs.knox for people to use.

emhastings committed:
Updated links/buttons based on help page

This commit updates the helpURLs for the blocks, and changes the
appearance of the submit button.  It still does not actually submit
anything.

emhastings committed:
Merge pull request #1 from jspacco/master

Merging Jaime changes

jspacco commented:
I started to integrate Blockly into the rest of the application, but I
haven't finished it. I need to figure out how to get the submit button to
produce the json, and then upload the javascript with the form. I don't
think this will be hard but I need to put some time into it.

To get Blockly integrated, I added a blocks.jsp file, and then changed
build.xml to copy all of the blockly stuff into the web folder of the jar
archive, where the Jetty server runs from. This seemed to work totally
fine, and I think it means that Blockly will just magically upload JSON
into Minecraft.

Oh, one thought: I know that the blocks don't let you create bad syntax,
but what happens if there is no "create turtle" block? We may need to
generate some additional error messages, or put errors in the JSON, or
something like that.

Dan Garcia from Berkeley had a really interesting 3D Turtles app that runs
through Snap, their version of Scratch. It's still in Beta, and they
weren't talking about it like Minecraft, but it was kind of similar. So
this idea is a good idea based on well-known people working on vaguely
similar things (this is basically "proof by eminent authority").

Jaime

On Fri, Mar 4, 2016 at 2:01 PM, Emily Hastings notifications@github.com
wrote:

> The Blocky documentation is done, and this commit updates the helpURLs for
> the blocks reflecting this. It also changes the appearance of the submit
> button. It still does not actually submit
> anything-- Jaime, I think you said last time we met that you were going to
> work on this?
> 
> When we're done with that, we also have to remember to put a download link
> for the Blockly stuff on the Google site with the other downloads, and
> 
> ## probably put our local version up on cs.knox for people to use.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/jspacco/Knoxcraft/pull/5
> Commit Summary
> - Updated links/buttons based on help page
> 
> File Changes
> - _M_ blockly/blocks/turtle.js
>   https://github.com/jspacco/Knoxcraft/pull/5/files#diff-0 (17)
> - _M_ blockly/index.html
>   https://github.com/jspacco/Knoxcraft/pull/5/files#diff-1 (5)
> 
> Patch Links:
> - https://github.com/jspacco/Knoxcraft/pull/5.patch
> - https://github.com/jspacco/Knoxcraft/pull/5.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jspacco/Knoxcraft/pull/5.

emhastings commented:
The code the runCode() method runs produces the json and then shows it with
an alert, so you could probably change the code generator for the init
block to submit instead of alert, or else append submit stuff to the code
before you run it in runCode(). I think, anyway.  I'm not sure of all the
details of submitting.

When there is no turtle init block, you get an error when you try to run
the code, because that's where the json variable declaration is.  We could
add a check for that when it tries to run the code, to get a more
user-friendly error message.

On Mon, Mar 7, 2016 at 10:20 AM, Jaime Spacco notifications@github.com
wrote:

> I started to integrate Blockly into the rest of the application, but I
> haven't finished it. I need to figure out how to get the submit button to
> produce the json, and then upload the javascript with the form. I don't
> think this will be hard but I need to put some time into it.
> 
> To get Blockly integrated, I added a blocks.jsp file, and then changed
> build.xml to copy all of the blockly stuff into the web folder of the jar
> archive, where the Jetty server runs from. This seemed to work totally
> fine, and I think it means that Blockly will just magically upload JSON
> into Minecraft.
> 
> Oh, one thought: I know that the blocks don't let you create bad syntax,
> but what happens if there is no "create turtle" block? We may need to
> generate some additional error messages, or put errors in the JSON, or
> something like that.
> 
> Dan Garcia from Berkeley had a really interesting 3D Turtles app that runs
> through Snap, their version of Scratch. It's still in Beta, and they
> weren't talking about it like Minecraft, but it was kind of similar. So
> this idea is a good idea based on well-known people working on vaguely
> similar things (this is basically "proof by eminent authority").
> 
> Jaime
> 
> On Fri, Mar 4, 2016 at 2:01 PM, Emily Hastings notifications@github.com
> wrote:
> 
> > The Blocky documentation is done, and this commit updates the helpURLs
> > for
> > the blocks reflecting this. It also changes the appearance of the submit
> > button. It still does not actually submit
> > anything-- Jaime, I think you said last time we met that you were going
> > to
> > work on this?
> > 
> > When we're done with that, we also have to remember to put a download
> > link
> > for the Blockly stuff on the Google site with the other downloads, and
> > 
> > ## probably put our local version up on cs.knox for people to use.
> > 
> > You can view, comment on, or merge this pull request online at:
> > 
> > https://github.com/jspacco/Knoxcraft/pull/5
> > Commit Summary
> > - Updated links/buttons based on help page
> > 
> > File Changes
> > - _M_ blockly/blocks/turtle.js
> >   https://github.com/jspacco/Knoxcraft/pull/5/files#diff-0 (17)
> > - _M_ blockly/index.html
> >   https://github.com/jspacco/Knoxcraft/pull/5/files#diff-1 (5)
> > 
> > Patch Links:
> > - https://github.com/jspacco/Knoxcraft/pull/5.patch
> > - https://github.com/jspacco/Knoxcraft/pull/5.diff
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/jspacco/Knoxcraft/pull/5.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jspacco/Knoxcraft/pull/5#issuecomment-193325243.

emhastings committed:
More helpful error message for missing init blocks

This commit adds a check for the init block, and produces a more
user-friendly error message when it is missing.
